server/base/src/main/java/org/apache/accumulo/server/util/checkCommand/SystemConfigCheckRunner.java

```diff
@@ -18,22 +18,230 @@
+    // not running, alerts the user that the process is not running which may or may not be expected
+    for (ServerProcess proc : serverProcesses) {
+      log.trace("Looking for {} lock(s)...", proc);
+      switch (proc) {
+        case MANAGER:
+          // essential process
+          status = checkLock(zkRoot + Constants.ZMANAGER_LOCK, proc, true, zs, status);
+          break;
+        case GC:
+          // essential process
+          status = checkLock(zkRoot + Constants.ZGC_LOCK, proc, true, zs, status);
+          break;
+        case TSERVER:
+          // essential process(es)
+          final var tservers = TabletMetadata.getLiveTServers(context);
+          if (tservers.isEmpty()) {
+            log.warn("Did not find any running tablet servers!");
+            status = Admin.CheckCommand.CheckStatus.FAILED;
+          }
+          break;
+        case COMPACTION_COORDINATOR:
+          // nonessential process
+          status = checkLock(zkRoot + Constants.ZCOORDINATOR_LOCK, proc, false, zs, status);
+          break;
+        case COMPACTOR:
+          // nonessential process(es)
+          if (compactors.isEmpty()) {
+            log.debug("No compactors appear to be running... This may or may not be expected");
+          }
+          for (String compactor : compactors) {
+            // for each running compactor, ensure a zk lock exists for it
+            boolean checkedLock = false;
+            String compactorQueuesPath = zkRoot + Constants.ZCOMPACTORS;
+            var compactorQueues = zrw.getChildren(compactorQueuesPath);
+            // find the queue the compactor is in
+            for (var queue : compactorQueues) {
+              String compactorQueuePath = compactorQueuesPath + "/" + queue;
+              String lockPath = compactorQueuePath + "/" + compactor;
+              if (zrw.exists(lockPath)) {
+                status = checkLock(lockPath, proc, true, zs, status);
+                checkedLock = true;
+                break;
+              }
+            }
+            if (!checkedLock) {
+              log.warn("Did not find a ZooKeeper lock for the compactor {}!", compactor);
+              status = Admin.CheckCommand.CheckStatus.FAILED;
+            }
+          }
+          break;
+        case MONITOR:
+          // nonessential process
+          status = checkLock(zkRoot + Constants.ZMONITOR_LOCK, proc, false, zs, status);
+          break;
+        case SCAN_SERVER:
+          // nonessential process(es)
+          if (sservers.isEmpty()) {
+            log.debug("No scan servers appear to be running... This may or may not be expected");
+          }
+          for (String sserver : sservers) {
+            status =
+                checkLock(zkRoot + Constants.ZSSERVERS + "/" + sserver, proc, true, zs, status);
+          }
+          break;
+        default:
+          throw new IllegalStateException("Unhandled case: " + proc);
+      }
+    }
```

Comment:
I imagine some of the non essential processes here in 3.x are essential in 4.x. I believe COMPACTION_COORDINATOR, COMPACTOR, SCAN_SERVER are all essential in 4.x? Is this correct? Will need to change this when merging into 4.x if so.